src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+	    {
+	      SLEEP_MILISEC (0, 10);
+	      retry_count++;
+	      goto retry;
+	    }
+	  ddl_log_fd = NULL;
+	}
+    }
+
+  return ddl_log_fd;
+}
+
+#if defined(WINDOWS)
+static void
+unix_style_path (char *path)
+{
+  char *p = NULL;
+  for (p = path; *p; p++)
+    {
+      if (*p == '\\')
+	*p = '/';
+    }
+}
+#endif /* WINDOWS */
+
+static int
+cub_create_dir_log (const char *new_dir)
+{
+  char *p, path[PATH_MAX] = { 0 };
+
+  if (new_dir == NULL)
+    return -1;
+
+  strcpy (path, new_dir);
+  trim (path);
+
+
+#if defined(WINDOWS)
+  unix_style_path (path);
+#endif /* WINDOWS */
+
+  p = path;
+#if defined(WINDOWS)
+  if (path[0] == '/')
+    p = path + 1;
+  else if (strlen (path) > 3 && path[2] == '/')
+    p = path + 3;
+#else /* WINDOWS */
+  if (path[0] == '/')
+    {
+      p = path + 1;
+    }
+#endif /* WINDOWS */
+
+  while (p != NULL)
+    {
+      p = strchr (p, '/');
+      if (p == NULL)
+	return 0;
+
+      if (p != NULL)
+	*p = '\0';
+
+      if (access (path, F_OK) < 0)
+	{
+	  if (mkdir (path, 0777) < 0)
+	    {
+	      return -1;
+	    }
+	}
+      if (p != NULL)
+	{
+	  *p = '/';
+	  p++;
+	}
+    }
+  return 0;
+}
+
+static int
+cub_get_current_time (char *buf, size_t size)
+{
+  struct tm at_tm;
+  int len = 0;
+  time_t t = time (NULL);
+
+  localtime_r (&t, &at_tm);
+  at_tm.tm_year += 1900;
+  at_tm.tm_mon += 1;
+  len =
+    snprintf (buf, size, "%04d%02d%02d_%02d%02d%02d", at_tm.tm_year, at_tm.tm_mon, at_tm.tm_mday, at_tm.tm_hour,
+	      at_tm.tm_min, at_tm.tm_sec);
+
+  return len;
+}
+
+static int
+cub_get_time_string (char *buf, struct timeval *time_val)
+{
+  struct tm tm, *tm_p;
+  time_t sec;
+  int millisec;
+
+  if (buf == NULL)
+    {
+      return 0;
+    }
+
+  if (time_val == NULL)
+    {
+      struct timeb tb;
+
+      /* current time */
+      (void) ftime (&tb);
+      sec = tb.time;
+      millisec = tb.millitm;
+    }
+  else
+    {
+      sec = time_val->tv_sec;
+      millisec = time_val->tv_usec / 1000;
+    }
+
+  tm_p = localtime_r (&sec, &tm);
+  tm.tm_mon++;
+
+  buf[0] = ((tm.tm_year % 100) / 10) + '0';
+  buf[1] = (tm.tm_year % 10) + '0';
+  buf[2] = '-';
+  buf[3] = (tm.tm_mon / 10) + '0';
+  buf[4] = (tm.tm_mon % 10) + '0';
+  buf[5] = '-';
+  buf[6] = (tm.tm_mday / 10) + '0';
+  buf[7] = (tm.tm_mday % 10) + '0';
+  buf[8] = ' ';
+  buf[9] = (tm.tm_hour / 10) + '0';
+  buf[10] = (tm.tm_hour % 10) + '0';
+  buf[11] = ':';
+  buf[12] = (tm.tm_min / 10) + '0';
+  buf[13] = (tm.tm_min % 10) + '0';
+  buf[14] = ':';
+  buf[15] = (tm.tm_sec / 10) + '0';
+  buf[16] = (tm.tm_sec % 10) + '0';
+  buf[17] = '.';
+  buf[20] = (millisec % 10) + '0';
+  millisec /= 10;
+  buf[19] = (millisec % 10) + '0';
+  millisec /= 10;
+  buf[18] = (millisec % 10) + '0';
+  buf[21] = '\0';
+
+  return 21;
+}
+
+static void
+cub_remove_char (char *string, char ch)
+{
+  for (; *string != '\0'; string++)
+    {
+      if (*string == ch)
+	{
+	  strcpy (string, string + 1);
+	  string--;
+	}
+    }
+}
+
+int
```

Comment:
You used,
#1166: return true;
#1171: return false;
then what about declarer return type of this function as '**bool**'

Reply:
Thank you, i will change code as your recommend.